php/class-post-type.php

```diff
@@ -549,4 +548,37 @@ public function filter_user_has_cap( $allcaps, $caps ) {
 
 		return $allcaps;
 	}
+
+	/**
+	 * Publish snapshot changes when snapshot post is being published.
+	 *
+	 * @param int      $post_id Post id.
+	 * @param \WP_Post $post Post object.
+	 */
+	public function publish_snapshot( $post_id, $post ) {
+		if ( ! is_a( $this->snapshot_manager->customize_manager, '\WP_Customize_Manager' ) ) {
+			require_once( ABSPATH . WPINC . '/class-wp-customize-manager.php' );
+			$this->snapshot_manager->customize_manager = new \WP_Customize_Manager();
+		}
+		$snapshot_content = json_decode( $post->post_content, true );
+		if ( json_last_error() || ! is_array( $snapshot_content ) ) {
+			return;
+		}
+		$snapshot_values = array_filter(
+			wp_list_pluck( $snapshot_content, 'value' ),
+			function( $value ) {
+				return ! is_null( $value );
+			}
+		);
+
+		/** This action is documented in wp-includes/class-wp-customize-manager.php */
+		do_action( 'customize_save', $this->snapshot_manager->customize_manager );
+
+		foreach ( $snapshot_values as $setting_id => $setting_params ) {
+			$this->snapshot_manager->customize_manager->set_post_value( $setting_id, $setting_params );
+			$this->snapshot_manager->customize_manager->add_setting( $setting_id, $setting_params );
+			$setting = $this->snapshot_manager->customize_manager->get_setting( $setting_id );
```

Comment:
@PatelUtkarsh you'll need to make sure that `$setting` is not `null`. Also, as noted in #15, you'll need to make sure you do: `$setting->capability = 'exist'` to ensure that the setting save will always be allowed. This is safe because a setting only be written to the snapshot by a user who has the capability to begin with, so we know at this point it will be safe to save.